kernel/hashlib.h

```diff
@@ -147,15 +137,14 @@ class HasherDJB32 {
 
 using Hasher = HasherDJB32;
 
-template<typename T>
-struct hash_top_ops {
-	static inline bool cmp(const T &a, const T &b) {
-		return hash_ops<T>::cmp(a, b);
-	}
-	static inline Hasher hash(const T &a) {
-		return hash_ops<T>::hash_into(a, Hasher());
-	}
-};
+// Boilerplate compressor for trivially implementing
+// top-level hash method with hash_into
+#define HASH_TOP_LOOP_FST [[nodiscard]] static inline Hasher hash
```

Comment:
Why not put all of the function into one macro, so that you would write e.g. `DEFAULT_HASH_TOP(std::pair<P, Q> a)` to use it?

Reply:
I tried doing this the simplest possible way I could think of, passing a type into a macro argument, and it wasn't valid preprocessor code for some reason. We can change this later